backend/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Controller,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Inject,
+  Request,
+  Get,
+  HttpStatus,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthService } from './auth.service';
+import { RequestWithRoles, RolesRequest } from './interface';
+import { AppLogger } from '../logger/logger.service';
+
+@Controller('auth')
+@ApiTags('Auth API')
+@UseInterceptors(ClassSerializerInterceptor)
+export class AuthController {
+  constructor(
+    @Inject(AuthService) private readonly authService: AuthService,
+    private readonly logger: AppLogger,
+  ) {
+    this.logger.setContext(AuthController.name);
+  }
+
+  @ApiOperation({
+    summary: 'Verify the role and sent user data',
+    description: 'Returns the user data with the request',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+  })
+  @Get()
+  async checkUserPermissions(
```

Comment:
I'd rename, at this point, we're querying user information. So getUserInfo? Or something

Reply:
copying to address in subsequent pr